### PR TITLE
Name changes from ViewStore -> Store

### PR DESCRIPTION
**Pull request by Pearapps**
## What it Does

Updates the `ViewStore` protocol to be named `Store`, and various other properties to indicate the explicit departure from only managing a view's state.

Also adding `ScopedStore` to allow for easier composition of `Store`s

## How I Tested

- Ran through the app and checked the ViewStore tab  
- Ran tests after updating them

### CODE REVIEW

File: Sources/Store/ScopedStore.swift [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    public init(initial: State, viewStatePub: some Publisher<State, Never>, action: @escaping (Action) -> Void) {
+        state = initial
+        self.action = action
+        viewStatePub.assign(to: &$state)
+    }
+
+    public func send(_ action: Action) {
+        self.action(action)
+    }
+
```
Comment (mliberatore) [inline Sources/Store/ScopedStore.swift]: let's collapse this space
```suggestion
```

File: Sources/Store/ScopedStore.swift [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        return $state.eraseToAnyPublisher()
+    }
+
+    private let action: (Action) -> Void
+    
+    /// Initializes a new `ScopedStore`
+    /// - Parameters:
+    ///   - initial: The initial state for this `Store`, likely a copy of whatever the current sub-store's state is now (see the `scoped` function on the `Store` extension for an example)
+    ///   - statePub: The publisher that allows this `ScopedStore` to get the lastest copy of the sub-store's state.
+    ///   - action: A closure to let you pass actions back to a parent `Store`. (see the `scoped` function on the `Store` extension for an example of embedding these into a "sub-action" of a parent Store to forward to a sub-store)
```
Comment (mliberatore) [inline Sources/Store/ScopedStore.swift:26]: Minor punctuation pass
```suggestion
    /// Initializes a new `ScopedStore`.
    /// - Parameters:
    ///   - initial: The initial state for this `Store`, likely a copy of whatever the current sub-store's state is now (see the `scoped` function on the `Store` extension for an example).
    ///   - statePub: The publisher that allows this `ScopedStore` to get the lastest copy of the sub-store's state.
    ///   - action: A closure to let you pass actions back to a parent `Store`. (see the `scoped` function on the `Store` extension for an example of embedding these into a "sub-action" of a parent Store to forward to a sub-store).
```

File: Sources/Store/ScopedStore.swift [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    ///     // Store logic and properties to manage some state
+    /// }
+    ///```
+    ///
+    /// - Parameters:
+    ///   - stateKeyPath: The keypath to the property on the Parent's `State` that is managed by the substore.
+    ///   - actionCasePath: The case path to an action on the Parent's `Store` that has the substore's action as the associated value that forwards to the substore.
+    /// - Returns: A `Store` that is scoped to the specified state and action.
+    func scoped<Substate, Subaction>(stateKeyPath: KeyPath<State, Substate>, actionCasePath: CasePath<Action, Subaction>) -> any Store<Substate, Subaction> {
+        return ScopedStore(initial: state[keyPath: stateKeyPath], statePub: publishedState.map(stateKeyPath), action: { self.send(actionCasePath.embed($0)) })
```
Comment (mliberatore) [inline Sources/Store/ScopedStore.swift:80]: Can we please do a find+replace from `statePub` → `statePublisher`?

**Review by mliberatore — COMMENTED** [review]
Quick pass on minor things

File: ExampleTests/StoreTests.swift [hunk 4]
```diff
@@ -1,5 +1,5 @@
 //
-//  ViewStoreTests.swift
+//  StoreTests.swift
 //  ViewStoreTests
```
Comment (Cordavi) [inline ExampleTests/StoreTests.swift:3]: I think this can be removed

**Review by mliberatore — APPROVED** [review]

**Pull request merged**